fasta/fasta.go

```diff
@@ -1,129 +1,156 @@
+			}
+			answer.Seq = append(answer.Seq, dna.StringToBases(line)...)
+		}
+
+		peek, err = fileio.EasyPeekReal(file, 1)
+		if errors.Is(err, io.EOF) {
+			return answer, doneReading // we could return true here, but we do not in order to match the behavior of other Next functions.
+		} else {
+			exception.PanicOnErr(err)
+		}
+
+		if peek[0] == '>' {
+			return answer, doneReading
+		}
 	}
+	return answer, doneReading
 }
 
-func WriteToFileHandle(file io.Writer, records []*Fasta, lineLength int) {
-	var err error
-	for _, rec := range records {
-		_, err = fmt.Fprintf(file, ">%s\n", rec.Name)
-		common.ExitIfError(err)
-		for i := 0; i < len(rec.Seq); i += lineLength {
-			if i+lineLength > len(rec.Seq) {
-				_, err = fmt.Fprintf(file, "%s\n", dna.BasesToString(rec.Seq[i:]))
-				common.ExitIfError(err)
-			} else {
-				_, err = fmt.Fprintf(file, "%s\n", dna.BasesToString(rec.Seq[i:i+lineLength]))
-				common.ExitIfError(err)
-			}
+// ToMap converts the a slice of fasta records (e.g. the output of the Read function)
+// to a map of sequences keyed to the sequences name.
+func ToMap(ref []*Fasta) map[string][]dna.Base {
+	m := make(map[string][]dna.Base)
+	for i := range ref {
+		_, ok := m[ref[i].Name]
+		if !ok {
+			m[ref[i].Name] = ref[i].Seq
+		} else {
+			log.Panicf("%s used for multiple fasta records. record names must be unique.", ref[i].Name)
 		}
 	}
+	return m
 }
 
+// Write a fasta to input filename. Output fastas have line length of 50.
 func Write(filename string, records []*Fasta) {
 	lineLength := 50
 	file := fileio.EasyCreate(filename)
-	defer file.Close()
-
 	WriteToFileHandle(file, records, lineLength)
+	exception.PanicOnErr(file.Close())
+}
+
+// WriteToFileHandle writes a slice of fasta records to a given io.Writer instead of creating
+// a new io.Writer as is done in the Write function.
+func WriteToFileHandle(file io.Writer, records []*Fasta, lineLength int) {
+	for _, rec := range records {
+		WriteFasta(file, rec, lineLength)
+	}
 }
 
+// WriteFasta writes a single fasta record to an io.Writer.
 func WriteFasta(file io.Writer, rec *Fasta, lineLength int) {
 	var err error
 	_, err = fmt.Fprintf(file, ">%s\n", rec.Name)
-	common.ExitIfError(err)
+	exception.PanicOnErr(err)
 	for i := 0; i < len(rec.Seq); i += lineLength {
 		if i+lineLength > len(rec.Seq) {
 			_, err = fmt.Fprintf(file, "%s\n", dna.BasesToString(rec.Seq[i:]))
-			common.ExitIfError(err)
+			exception.PanicOnErr(err)
 		} else {
 			_, err = fmt.Fprintf(file, "%s\n", dna.BasesToString(rec.Seq[i:i+lineLength]))
-			common.ExitIfError(err)
+			exception.PanicOnErr(err)
 		}
 	}
 }
 
-func WriteGroups(filename string, groups [][]*Fasta) error {
-	lineLength := 50
-	file, err := os.Create(filename)
-	if err != nil {
-		return err
+// SplitWrite writes each fasta record to its own file.
+func SplitWrite(filename string, records []*Fasta) {
+	for _, rec := range records {
+		Write(filename+"_"+rec.Name+"_"+".fasta", []*Fasta{rec})
```

Comment:
On second pass, I think it would be best to delete this function. Currently no function/cmd in gonomics calls SplitWrite and it seems to me like it would fit better in a main function, that way each tool can define record names as they like.